### PR TITLE
backend/repository: assume no v2 on unexpected status

### PR DESCRIPTION
**Pull request by lucab**
This commit changes version detection behavior to align with
Registry v2 specs:

```
If 404 Not Found response status, or other unexpected status,
is returned, the client should proceed with the assumption
that the registry does not implement V2 of the API.
```

Reference: https://docs.docker.com/registry/spec/api/#/api-version-check

Fixes https://github.com/appc/docker2aci/issues/85

### CODE REVIEW

File: lib/internal/backend/repository/repository.go [hunk 1]
```diff
@@ -108,27 +108,27 @@ func (rb *RepositoryBackend) BuildACI(layerIDs []string, dockerURL *types.Parsed
...
+			// and clients MAY fallback to v1 if missing, as done here.
 			ok = hdr.Get("Docker-Distribution-API-Version") == "registry/2.0"
 		}
 		return ok, nil
-	case http.StatusNotFound:
+	default:
 		return false, nil
 	}
-
-	return false, fmt.Errorf("unexpected http code: %d", statusCode)
```
Comment (s-urbaniak) [inline lib/internal/backend/repository/repository.go]: I think this is a compilation failure, since there is nothing returned.

Comment (lucab) [inline lib/internal/backend/repository/repository.go]: Local build+test seems fine (go 1.6.3), but I see over-zealous checkers may get confused. I'll switch to a false-return at the bottom.

**Comment by lucab** [comment]
@s-urbaniak rebased, PTAL.

**Comment by jonboulle** [comment]
LGTM

**Pull request merged**